File: frontend/src/api/companionRole.ts
import request, { PageResult } from './request'

export interface CompanionRole {
  id?: number
  roleName: string
  roleIdentity?: string
  avatarUrl?: string
  themeColor?: string
  background?: string
  personality?: string
  expertise?: string
  hobbies?: string
  speakingStyle?: string
  scenario?: string
  companionGoal?: string
  boundaries?: string
  customPrompt?: string
  tags?: string
  defaultRole?: boolean
  status?: string
}

export function listCompanionRoles(params: any = {}) {
  return request.get<any, PageResult<CompanionRole>>('/companion-roles', { params })
}

export function getActiveCompanionRole() {
  return request.get<any, CompanionRole | null>('/companion-roles/active')
}

export function createCompanionRole(data: CompanionRole) {
  return request.post<any, CompanionRole>('/companion-roles', data)
}

export function updateCompanionRole(id: number, data: Partial<CompanionRole>) {
  return request.put<any, CompanionRole>(`/companion-roles/${id}`, data)
}

export function deleteCompanionRole(id: number) {
  return request.delete<any, void>(`/companion-roles/${id}`)
}

export function setDefaultCompanionRole(id: number) {
  return request.post<any, CompanionRole>(`/companion-roles/${id}/default`)
}

File: backend/src/main/java/com/edustudio/module/companion/entity/CompanionRole.java
package com.edustudio.module.companion.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.edustudio.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("ai_companion_role")
public class CompanionRole extends BaseEntity {

    private Long userId;

    private String roleName;

    private String roleIdentity;

    private String avatarUrl;

    private String themeColor;

    private String background;

    private String personality;

    private String expertise;

    private String hobbies;

    private String speakingStyle;

    private String scenario;

    private String companionGoal;

    private String boundaries;

    private String customPrompt;

    private String tags;

    @TableField("is_default")
    private Boolean defaultRole;

    private String status;
}
